lib/pf/api.pm

```diff
@@ -522,13 +519,32 @@ sub distant_download_configfile : Public {
     my %data = ( conf_file => $file );
     my $apiclient = pf::api::jsonrpcclient->new(host => $postdata{from}, proto => 'https');
     my ($result) = $apiclient->call( 'download_configfile', %data );
-    open(my $fh, '>', $file);
-    print $fh $result;
-    close($fh);
-    `chown pf.pf $file`;
-
+    _safe_file_update($file, $result);
     return 1;
+}
+
+=head2 _safe_file_update
 
+This safely modifies the contents of a file using a rename
+
+=cut
+
+sub _safe_file_update {
+    my ($file, $contents) = @_;
+    my ($volume, $dir, $filename) = File::Spec->splitpath($file);
+    $dir = '.' if $dir eq '';
+    # Creates a new file in the same directory to ensure it is on the same filesystem
+    my $temp = File::Temp->new(DIR => $dir) or die "cannot create temp file in $dir";
+    syswrite $temp, $contents;
+    $temp->flush;
+    close $temp;
+    unless( rename ($temp->filename, $file) ) {
+        my $logger = pf::log::get_logger;
+        $logger->error("cannot save contents to $file '$!'");
+        die "cannot save contents to $file";
+    }
+    $temp->unlink_on_destroy(0);
+    `chown pf.pf $file`;
```

Comment:
Yes, I would suggest to base off that PR and cite it as a dependency as both will be part of the same release